### PR TITLE
//boost each input pixel by 64, the alginment scoring seems to work b…

### PR DESCRIPTION
**Pull request by KillerInk**
…etter with higher values

//we loose 64 in highlights but on 12-14bit input its negligible

#21

### CODE REVIEW

**Comment by KillerInk** [comment]
this is bevor https://github.com/timothybrooks/hdr-plus/pull/23/commits/cccf18eb61f1a08a8dde19a33f934c31031ba687

bevor-3-8

this after.

File: src/align.h [hunk 1]
```diff
@@ -4,8 +4,8 @@
 #define T_SIZE 32           // Size of a tile in the bayer mosaiced image
 #define T_SIZE_2 16         // Half of T_SIZE and the size of a tile throughout the alignment pyramid
 
-#define MIN_OFFSET -168     // Min total alignment (based on three levels and downsampling by 4)
-#define MAX_OFFSET 126      // Max total alignment. Differs from MIN_OFFSET because total search range is 8 for better vectorization
+#define MIN_OFFSET -256     // Min total alignment (based on three levels and downsampling by 4)
+#define MAX_OFFSET 256      // Max total alignment. Differs from MIN_OFFSET because total search range is 8 for better vectorization
```
Comment (timothybrooks) [inline src/align.h]: What are the performance impacts of this change? The search region is kept somewhat small such that it vectorizes cleanly. Does the processing run at a similar speed?

File: src/merge.cpp [hunk 2]
```diff
@@ -49,15 +49,14 @@ Func merge_temporal(Image<uint16_t> imgs, Func alignment) {
 
     // constants for determining strength and robustness of temporal merge
 
-    float factor = 8.f;                         // factor by which inverse function is elongated
-    int min_dist = 10;                          // pixel L1 distance below which weight is maximal
-    int max_dist = 300;                         // pixel L1 distance above which weight is zero
+    int min_dist = 4;                          // pixel L1 distance below which weight is maximal
+    int max_dist = 128;                         // pixel L1 distance above which weight is zero
```
Comment (timothybrooks) [inline src/merge.cpp]: What raw images are you using to adjust this tuning? I tuned for the Canon 5D III. I am hesitant to change tuning without testing on multiple raw, underexposed bursts, as intended. If you would like, I could send you raw images for testing.

File: src/align.h (hunk 1 above)
Comment (KillerInk) [reply src/align.h]: i dont think it has a big performance impact. at least i dont feel it. but time is relative when you sit hours infront of it and diff images ;) I will trace the time it takes to process, bevor and after.
What i have noticed is that good tiles get sorted out when the camera got shaked to heavy and bad tiles that match that range gets more weighted wich can result in ghosting.

**Comment by KillerInk** [comment]
> Rather than changing the black point and increasing the actual image brightness, this change would better live in the alignment code, as an offset used when calculating correlations. It will have the same functionality, but keep the image and black point values intact, which is useful incase these values are used in other parts of the processing (e.g. computing hot pixels or bias introduced by black level, computing noise levels, etc.).

it would fit there better. but due the fact that the black lvl later gets used only int the black_white_level Func its removed from later processing. hotpixel will fail anyway because they get merged.
and for the contrast func a different black lvl gets used, but im not sofar yet. then only thing i changed on the finish part is to work with the ImageParam.

**Pull request closed without merging**